### PR TITLE
feat(api): Put CtAnonymousExecutable under CtExecutable.

### LINKED ISSUE #426: put CtAnonymousExecutable under CtExecutable
appropriate for the 5.0 that is in preparation

### PR DESCRIPTION
**Pull request by GerardPaligot**
Closes #426

### CODE REVIEW

**Comment by GerardPaligot** [comment]
@monperrus PR ok.

File: src/main/java/spoon/support/reflect/declaration/CtAnonymousExecutableImpl.java [hunk 1]
```diff
@@ -112,4 +103,63 @@ public boolean hasModifier(ModifierKind modifier) {
...
+	}
+
+	@Override
+	public <T extends CtNamedElement> T setSimpleName(String simpleName) {
+		throw new UnsupportedOperationException("You can't have a name in a anonymous executable");
+	}
+
+	@Override
+	public CtTypeReference<Void> getType() {
+		throw new UnsupportedOperationException("An anonymous executable isn't typed");
```
Comment (monperrus) [inline src/main/java/spoon/support/reflect/declaration/CtAnonymousExecutableImpl.java]: should return CtTypeReference.VOID instead

File: src/main/java/spoon/support/reflect/declaration/CtAnonymousExecutableImpl.java [hunk 2]
```diff
@@ -112,4 +103,63 @@ public boolean hasModifier(ModifierKind modifier) {
...
+	}
+
+	@Override
+	public boolean removeThrownType(CtTypeReference throwType) {
+		throw new UnsupportedOperationException("You can't have throw types in a anonymous executable");
+	}
+
+	@Override
+	public String getSimpleName() {
+		throw new UnsupportedOperationException("You can't have a name in a anonymous executable");
```
Comment (monperrus) [inline src/main/java/spoon/support/reflect/declaration/CtAnonymousExecutableImpl.java]: breaks clieant code, should return empty string or "<static>" (inspired from "<init>")

**Comment by monperrus** [comment]
went too fast on this one, it breaks client code, see line comment

**Pull request merged**